tests/unit/api/BackbeatRequest.spec.js

```diff
@@ -0,0 +1,41 @@
+const assert = require('assert');
+
+const BackbeatRequest = require('../../../lib/api/BackbeatRequest');
+
+function setRouteDetails(bbRequest, route) {
+    bbRequest.setRoute(route);
+    bbRequest._parseRoute();
```

Comment:
Instead of calling this private method here, shouldn't it be called automatically from `getRouteDetails()` (only if needed, i.e if not done before)?

Reply:
Ah, especially for unit tests, this would look confusing. Rewriting this to use a new instance of BackbeatRequest for each test